admin-dev/themes/new-theme/js/pages/product-preferences/catalog-mode-option-handler.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+const $ = window.$;
+
+class CatalogModeOptionHandler {
+  constructor() {
+    this.handle(0);
+
+    $('input[name="form[general][catalog_mode]"]').on('change', () => this.handle(600));
+  }
+
+  handle(fadeLength) {
+    const catalogModeVal = $('input[name="form[general][catalog_mode]"]:checked').val();
+    const catalogModeEnabled = parseInt(catalogModeVal);
+
+    let catalogOptions = $('.catalog-mode-option');
```

Comment:
All selectors must be extracted in a PageMap object in a separate file (to be used both by the page and E2E tests):

product-preferences-page-map.js:

```js
export default {
	catalogModeField: 'input[name="form[general][catalog_mode]',
	catalogModeOptions: '.catalog-mode-option',
	// ...
};
```

index.js
```js
import * as pageMap from 'product-preferences-page-map';

// ...
new CatalogModeOptionsHandler(pageMap);
```